### PR TITLE
Installation of knxd now uses knxd package from standard package repository

### LINKED ISSUE #651: Rewrite KNXD to use Debian package during installation
Installing optional component knxd fails on RPI3B+ with Buster.
This is probably an issue with upstream repos.

<code>
/bin/bash ../../libtool  --tag=CXX   --mode=link g++  -g -O2 -std=c++0x -Wno-subobject-linkage   -o findknxusb findknxusb.o -lusb-1.0 -lev -lfmt 
libtool: link: g++ -g -O2 -std=c++0x -Wno-subobject-linkage -o findknxusb findknxusb.o  -lusb-1.0 -lev -lfmt
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../src/libserver -I../../src/common -I/usr/include/libusb-1.0 -I/root/knxdbuild/fmt  -g -O2 -std=c++0x -Wno-subobject-linkage -MT usb.o -MD -MP -MF .deps/usb.Tpo -c -o usb.o usb.cpp
In file included from /usr/lib/gcc/arm-linux-gnueabihf/8/include-fixed/limits.h:194,
                 from /usr/lib/gcc/arm-linux-gnueabihf/8/include-fixed/syslimits.h:7,
                 from /usr/lib/gcc/arm-linux-gnueabihf/8/include-fixed/limits.h:34,
                 from /usr/include/libusb-1.0/libusb.h:62,
                 from usb.h:24,
                 from usb.cpp:23:
/root/knxdbuild/fmt/fmt/format.h: In member function ‘void fmt::internal::ArgFormatterBase<Impl, Char>::visit_char(int)’:
/root/knxdbuild/fmt/fmt/format.h:1825:20: error: expected unqualified-id before numeric constant
     const unsigned CHAR_WIDTH = 1;
                    ^~~~~~~~~~
[...few more warnings...]
make[3]: *** [Makefile:501: usb.o] Error 1
make[3]: Leaving directory '/root/knxdbuild/knxd/src/usb'
make[2]: *** [Makefile:394: all-recursive] Error 1
make[2]: Leaving directory '/root/knxdbuild/knxd/src'
make[1]: *** [Makefile:442: all-recursive] Error 1
make[1]: Leaving directory '/root/knxdbuild/knxd'
make: *** [Makefile:378: all] Error 2
Created symlink /etc/systemd/system/multi-user.target.wants/knxd.service → /lib/systemd/system/knxd.service.
Created symlink /etc/systemd/system/network-online.target.wants/knxd.service → /lib/systemd/system/knxd.service.
Removed /etc/systemd/system/multi-user.target.wants/hciuart.service.
Please reboot your device!
OK. Please restart your system now...
</code>

Actually this is a weird error caused by a name conflict between libfmt and glibc. Newer glibc includes define CHAR_WIDTH to 8, resulting in the code `const unsigned 8 = 1`.
See also https://github.com/fmtlib/fmt/pull/616
This is fixed in libfmt 5.0.0, though the script we download is using tags/3.0.0.

The question is how to get a hotfix for openhabian in here, either patching the header file or switching to a newer tag.

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Changed from building and installing latest source to installing the
knxd package provided by several distrubutions. Package is available
in Debian, Raspberry Pi OS, and Ubuntu (starting 20.04). Closes #651.
Added bats test case.

Signed-off-by: Holger Friedrich <mail@holger-friedrich.de>

### CODE REVIEW

File: functions/packages.bash [hunk 1]
```diff
@@ -396,27 +396,50 @@ find_setup() {
 ##    knxd_setup()
 ##
 knxd_setup() {
-  local introText="This will install and setup kndx (successor to eibd) as your EIB/KNX IP gateway and router to support your KNX bus system.\\n\\nThis routine was provided by 'Michels Tech Blog': https://bit.ly/3dzeoKh"
-  local successText="Setup was successful.\\n\\nPlease edit '/etc/default/knxd' to meet your interface requirements. For further information on knxd options, please type 'knxd --help'\\n\\nFurther details can be found under: https://bit.ly/3dzeoKh\\n\\nIntegration into openHAB 2 is described here: https://github.com/openhab/openhab/wiki/KNX-Binding"
+  local introText="This will install and setup kndx as your EIB/KNX IP gateway and router to support your KNX bus system.\\n\\nNOTE: You don't need this if you connect via an IP interfacte or router to your KNX installation. This package is to turn an USB or serial interface into an IP interface.\n\nNOTE: openHABian changed from building and installing latest source to installing the knxd package provided by several distrubutions."
```
Comment (mstormi) [inline functions/packages.bash:399]: will it also _setup_ knxd? If no remove from this text

2 typos:
interfacte 
distrubutions

**Review by mstormi — APPROVED** [review]
other than the text it LGTM

File: functions/packages.bash (hunk 1 above)
Comment (holgerfriedrich) [reply functions/packages.bash]: Ok, thanks. Wish I could find a spell checker for strings bash scripts....
Will merge once checks have completed.

**Comment by holgerfriedrich** [comment]
@mstormi is focal fossa always failing due to missing packages or did I introduce a problem here?
I have seen homegear failing because 20.04 dropped python-lzo package and telldus because of stretch repo (anyway, there is no libconfuse1 available....) -> shall we create new tickets for this or ignore?

**Comment by mstormi** [comment]
You refer to the BATS tests on Ubuntu ? It wasn't you. The test is new :) Issue #1161 is already open.
If you can fix any of these please do (I had started on the Telldus maybe try any of the others first).

**Pull request merged**